flocker/control/test/test_httpapi.py

```diff
@@ -1309,6 +1309,263 @@ def test_container_with_volume(self):
+            }
+        ))
+
+        saving.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers",
+            {
+                u"host": self.NODE_A,
+                u"name": u"mycontainer",
+                u"image": u"busybox"
+            }, CREATED
+        ))
+
+        return saving
+
+    def test_update_same_host(self):
+        """
+        An API request to update a named container's host to the same host
+        on which it is already running results in an unchanged configuration.
+        """
+        d = self._create_container()
+
+        d.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers/mycontainer",
+            {u"host": self.NODE_A}, OK
+        ))
+
+        def updated(_):
+            deployment = self.persistence_service.get()
+            expected = Deployment(
+                nodes={
+                    Node(
+                        hostname=self.NODE_A,
+                        applications=[
+                            Application(
+                                name='leavemealone',
+                                image=DockerImage.from_string('busybox'),
+                            ),
+                            Application(
+                                name='mycontainer',
+                                image=DockerImage.from_string('busybox')
+                            ),
+                        ]
+                    ),
+                    Node(hostname=self.NODE_B),
+                }
+            )
+            self.assertEqual(deployment, expected)
+
+        d.addCallback(updated)
+        return d
+
+    def test_update_new_host(self):
+        """
+        An API request to update a named container's host to a different host
+        results in an updated configuration.
+        """
+        d = self._create_container()
+
+        d.addCallback(lambda _: self.assertResponseCode(
+            b"POST", b"/configuration/containers/mycontainer",
+            {u"host": self.NODE_B}, OK
+        ))
+
+        def updated(_):
+            deployment = self.persistence_service.get()
+            expected = Deployment(
+                nodes={
+                    Node(
+                        hostname=self.NODE_A,
+                        applications=[
+                            Application(
+                                name='leavemealone',
+                                image=DockerImage.from_string('busybox'),
+                            ),
+                        ]
+                    ),
+                    Node(
+                        hostname=self.NODE_B,
+                        applications=[
+                            Application(
+                                name='mycontainer',
+                                image=DockerImage.from_string('busybox')
+                            ),
+                        ]
+                    ),
+                }
+            )
```

Comment:
Similarly, you might be able to grab the config before making the POST and then use a much shorter `transform` call to construct the expected value.